hummingbot/connector/derivative/bybit_perpetual/bybit_perpetual_user_stream_data_source.py

```diff
@@ -109,8 +116,30 @@ async def listen_for_user_stream(self, output: asyncio.Queue):
                 raise
             except Exception as ex:
                 self.logger().error(
-                    f"Unexpected error with Bybit Perpetual WebSocket connection. Retrying in 30 seconds. ({ex})",
+                    f"Unexpected error with Bybit Perpetual WebSocket connection on {url}."
+                    f" Retrying in 30 seconds. ({ex})",
                     exc_info=True
                 )
                 await ws.close()
                 await asyncio.sleep(30.0)
+
+    async def listen_for_user_stream(self, output: asyncio.Queue):
+        """
+        Subscribe to all required events and start the listening cycle.
+        """
+        tasks_future = None
+        try:
+            tasks = []
+            tasks.append(self._listen_for_user_stream_on_url(
+                url=bybit_perpetual_utils.wss_linear_private_url(self._domain),
+                output=output))
+            tasks.append(self._listen_for_user_stream_on_url(
+                url=bybit_perpetual_utils.wss_non_linear_private_url(self._domain),
+                output=output))
+
+            tasks_future = asyncio.gather(*tasks)
+            await tasks_future
```

Comment:
Technically speaking, the coroutine `_listen_for_user_stream_on_url()` does not end right? So `asyncio.gather()` does not really complete. I think we should use `safe_ensure_future()` instead and cancel when the bot stops.

Reply:
I am not sure that is a good idea. If we do that, the result would be that the task created in the tracker, the one calling listen_for_user_stream, would finish (because once it creates the future if would finish). But currently the whole architecture is designed for that task to be a live task, that should be canceled if something goes wrong. Using `gather` I am keeping that same behaviour.